chainercv/extensions/detection/detection_vis_report.py

```diff
@@ -210,62 +82,48 @@ def __call__(self, trainer):
         if not _available:
             return
 
-        for idx in self.indices:
-            formated_filename_base = osp.join(trainer.out, self.filename_base)
-            out_file = (formated_filename_base +
-                        '_idx={}'.format(idx) +
-                        '_iter={}'.format(trainer.updater.iteration) + '.jpg')
-
-            inputs = self.dataset[idx]
-            self._check_type_dataset(inputs)
-
-            pred_bbox, pred_label = forward(self.target, inputs[0],
-                                            forward_func=self.predict_func,
-                                            expand_dim=True)
-            if pred_bbox.ndim == 2:
-                # force output to have batch axis
-                pred_bbox = pred_bbox[None]
-                pred_label = pred_label[None]
-            self._check_type_model((pred_bbox, pred_label))
-            pred_bbox = pred_bbox[0]  # (B, R, 5) -> (R, 5)
-
-            vis_transformed = self.vis_transform(inputs)
-            self._check_type_vis_transformed(vis_transformed)
-            vis_img = vis_transformed[0]
-            raw_bbox = vis_transformed[1]
-
-            # start visualizing using matplotlib
-            fig = plot.figure()
-
-            ax_gt = fig.add_subplot(2, 1, 1)
-            ax_gt.set_title('ground truth')
-            label_names = getattr(self.dataset, 'labels', None)
-            vis_bbox(
-                vis_img, raw_bbox, label_names=label_names, ax=ax_gt)
-
-            ax_pred = fig.add_subplot(2, 1, 2)
-            ax_pred.set_title('prediction')
-
-            vis_bbox(vis_img, pred_bbox, label_names=label_names, ax=ax_pred)
-
-            plot.savefig(out_file)
-            plot.close()
-
-
-if __name__ == '__main__':
-    from chainercv.datasets import VOCDetectionDataset
-    from chainercv.utils.test_utils import ConstantReturnModel
-    import mock
-    import tempfile
-    train_data = VOCDetectionDataset(mode='train', year='2007')
-    _, bbox, label = train_data[3]
-
-    model = ConstantReturnModel((bbox[None], label[None]))
-
-    trainer = mock.MagicMock()
-    out_dir = tempfile.mkdtemp()
-    print('outdir ', out_dir)
-    trainer.out = out_dir
-    trainer.updater.iteration = 0
-    extension = DetectionVisReport([3], train_data, model)
-    extension(trainer)
+        if hasattr(self.iterator, 'reset'):
+            self.iterator.reset()
+            it = self.iterator
```

Comment:
I guess these evaluator like extensions are assumed to invoke `reset` at the beginning of `__call__`.

It looks fine now.